### PR TITLE
Update Sphinx docs to work with monorepository and pipenv

### LINKED ISSUE #44: Docs are not being built
Since the monorepo-ification the doc build is failing. Here's the log:

```
 Running Sphinx v1.6.6
making output directory...
WARN engine lerna@3.6.0: wanted: {"node":">= 6.9.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
WARN engine eslint@5.10.0: wanted: {"node":"^6.14.0 || ^8.10.0 || >=9.10.0"}

... many many many warnings


WARN engine @lerna/npm-run-script@3.6.0: wanted: {"node":">= 6.9.0"} (current: {"node":"4.2.6","npm":"3.5.2"})
npm WARN prefer global node-gyp@3.8.0 should be installed with -g
/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest

loading translations [en]... done
ERROR: Unable to find the source file or directory /home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest/node_modules/minim/lib
ERROR: Unable to find the source file or directory /home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest/node_modules/minim/lib/primitives
ERROR: Unable to find the source file or directory /home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest/node_modules/minim/lib/elements
ERROR: Unable to find the source file or directory /home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest/node_modules/minim/lib/serialisers
ERROR: Unable to find the source file or directory /home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest/node_modules/minim-api-description/lib/elements
ERROR: Unable to find the source file or directory /home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest/node_modules/minim-parse-result/lib

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx_js/jsdoc.py", line 38, in run_jsdoc
    doclets = load(temp)
  File "/home/docs/.pyenv/versions/3.6.2/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/home/docs/.pyenv/versions/3.6.2/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/home/docs/.pyenv/versions/3.6.2/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/docs/.pyenv/versions/3.6.2/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx/cmdline.py", line 305, in main
    opts.warningiserror, opts.tags, opts.verbosity, opts.jobs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx/application.py", line 234, in __init__
    self._init_builder()
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx/application.py", line 312, in _init_builder
    self.emit('builder-inited')
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx/application.py", line 489, in emit
    return self.events.emit(event, self, *args)
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx/events.py", line 79, in emit
    results.append(callback(*args))
  File "/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/envs/latest/lib/python3.6/site-packages/sphinx_js/jsdoc.py", line 40, in run_jsdoc
    raise SphinxError('jsdoc found no JS files in the directories %s. Make sure js_source_path is set correctly in conf.py. It is also possible (though unlikely) that jsdoc emitted invalid JSON.' % abs_source_paths)
sphinx.errors.SphinxError: jsdoc found no JS files in the directories ['/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest']. Make sure js_source_path is set correctly in conf.py. It is also possible (though unlikely) that jsdoc emitted invalid JSON.

Sphinx error:
jsdoc found no JS files in the directories ['/home/docs/checkouts/readthedocs.org/user_builds/api-elements-js/checkouts/latest']. Make sure js_source_path is set correctly in conf.py. It is also possible (though unlikely) that jsdoc emitted invalid JSON.

``

### PR DESCRIPTION
**Pull request by kylef**
I've also started building docs on CI so we can catch introduced problems similiar to Dredd setup.

- Docs now use pipenv
- Docs are built on CI
- See specific commits for further details

Closes #44

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -35,6 +35,22 @@ jobs:
...
+      - checkout
+      - <<: *install
+      - run:
+          working_directory: docs
+          command: pipenv install --dev
+      - run:
+          working_directory: docs
+          command: |
+            export PATH="$(yarn bin):$PATH"
+            pipenv run make html
```
Comment (honzajavorek) [inline .circleci/config.yml]: could be `pipenv run build`

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,6 +35,22 @@ jobs:
...
+  docs:
+    <<: *defaults
+    docker:
+      - image: circleci/python:3.6-node-browsers
+    steps:
+      - checkout
+      - <<: *install
+      - run:
+          working_directory: docs
+          command: pipenv install --dev
```
Comment (honzajavorek) [inline .circleci/config.yml]: currently the only dev dependency is autobuild, which isn't needed on CI, but depends whether you plan to add more dev deps

File: docs/conf.py [hunk 3]
```diff
@@ -9,7 +9,8 @@
 if IS_READTHEDOCS:
     docs_dir = os.path.dirname(__file__)
     project_dir = os.path.join(docs_dir, '..')
-    subprocess.check_call('npm install', cwd=project_dir, shell=True)
+    subprocess.check_call('npm install --global yarn', cwd=project_dir, shell=True)
```
Comment (honzajavorek) [inline docs/conf.py]: be aware this is very old npm and old node (I don't recall the versions now, but it caught me surprised)

File: .circleci/config.yml (hunk 2 above)
Comment (kylef) [reply .circleci/config.yml]: I wanted to look into linting and such, spell checking etc so plan was to iterate on that afterwards.

**Pull request merged**